src/main/java/org/folio/rest/impl/CopycatImpl.java

```diff
@@ -118,7 +129,16 @@ public void postCopycatImports(CopyCatImports entity, Map<String, String> okapiH
                   entity::getInternalIdentifier, () -> pattern);
               JsonMarc.embedPath(marc, pattern, entity.getInternalIdentifier());
             } else {
-              jobProfile = targetProfile.getCreateJobProfileId();
+              if (selectedJobProfile != null) {
+                List<String> createJobProfileIds = targetProfile.getCreateJobProfileIds();
+                if (createJobProfileIds != null
+                    && !createJobProfileIds.contains(selectedJobProfile)) {
+                  return Future.failedFuture("Invalid job profile id");
+                }
+                jobProfile = selectedJobProfile;
+              } else {
+                jobProfile = targetProfile.getDefaultCreateJobProfileId();
+              }
```

Comment:
Duplicating logic. Refactor to a separate function.

Reply:
Function refactored.